### ISSUE TITLE
Recursive loading with schema.load_schema is failing in 1.2.1

### ISSUE THREAD

**Issue by majensen** [fastavro/fastavro#490]
Using the example at https://fastavro.readthedocs.io/en/latest/schema.html (but adding 
```
    "namespace": "namespace"
```
to Parent.avsc), recursive loading works for fastavro 1.1.1, but fails for 1.2.1. (Py3.9, installed with homebrew on MacOS 10.14.6, in a venv)

See the comparison below (1.1.1 top; 1.2.1 bottom).
Thanks 

```
In [1]: import fastavro

In [2]: ??fastavro.__version__
Type:        str
String form: 1.1.1
Length:      5
Docstring:  
str(object='') -> str
str(bytes_or_buffer[, encoding[, errors]]) -> str

Create a new string object from the given object. If encoding or
errors is specified, then the object must expose a data buffer
that will be decoded using the given encoding and error handler.
Otherwise, returns the result of object.__str__() (if defined)
or repr(object).
encoding defaults to sys.getdefaultencoding().
errors defaults to 'strict'.

In [3]: fastavro.schema.load_schema("Parent.avsc");

In [4]: fastavro.schema.load_schema("Parent.avsc")
Out[4]: 
[{'type': 'record',
  'name': 'namespace.Child',
  'fields': [],
  '__fastavro_parsed': True,
  '__named_schemas': {'namespace.Child': {'type': 'record',
    'name': 'namespace.Child',
    'fields': []},
   'namespace.Parent': {'type': 'record',
    'name': 'namespace.Parent',
    'fields': [{'name': 'child', 'type': 'namespace.Child'}]}}},
 {'type': 'record',
  'name': 'namespace.Parent',
  'fields': [{'name': 'child', 'type': 'namespace.Child'}],
  '__fastavro_parsed': True,
  '__named_schemas': {'namespace.Child': {'type': 'record',
    'name': 'namespace.Child',
    'fields': []},
   'namespace.Parent': {'type': 'record',
    'name': 'namespace.Parent',
    'fields': [{'name': 'child', 'type': 'namespace.Child'}]}}}]

In [5]: quit
```

```
Python 3.9.0 (default, Dec  6 2020, 20:20:21) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.19.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import fastavro

In [2]: ??fastavro.__version__
Type:        str
String form: 1.2.1
Length:      5
Docstring:  
str(object='') -> str
str(bytes_or_buffer[, encoding[, errors]]) -> str

Create a new string object from the given object. If encoding or
errors is specified, then the object must expose a data buffer
that will be decoded using the given encoding and error handler.
Otherwise, returns the result of object.__str__() (if defined)
or repr(object).
encoding defaults to sys.getdefaultencoding().
errors defaults to 'strict'.

In [3]: fastavro.schema.load_schema("Parent.avsc");
---------------------------------------------------------------------------
UnknownType                               Traceback (most recent call last)
fastavro/_schema.pyx in fastavro._schema._load_schema()

fastavro/_schema.pyx in fastavro._schema.parse_schema()

fastavro/_schema.pyx in fastavro._schema._parse_schema()

fastavro/_schema.pyx in fastavro._schema.parse_field()

fastavro/_schema.pyx in fastavro._schema._parse_schema()

UnknownType: namespace.Child

During handling of the above exception, another exception occurred:

Exception                                 Traceback (most recent call last)
<ipython-input-3-8c930452e82a> in <module>
----> 1 fastavro.schema.load_schema("Parent.avsc");

fastavro/_schema.pyx in fastavro._schema.load_schema()

fastavro/_schema.pyx in fastavro._schema._load_schema()

fastavro/_schema.pyx in fastavro._schema._inject_schema()

fastavro/_schema.pyx in fastavro._schema._inject_schema()

Exception: Internal error; You should raise an issue in the fastavro github repository

In [4]: quit
```

**Comment by scottbelden** [comment]
Thanks for the report. I need to fix that example as it fails currently even without adding the `namespace` field to `Parent`. However, it looks like it should be pretty easy to resolve the issue you are seeing. I'll try to take a look soon but the holidays might delay it a bit.

**Comment by majensen** [comment]
@scottbelden Do you need something like the following at _schema_py.py:533 ? Seems to fail still when an enum is present.

```
        elif schema_type == "enum":
            return_schema, injected = _inject_schema(
                outer_schema["symbols"], inner_schema, namespace, is_injected
            )
            outer_schema["symbols"] = return_schema
            return outer_schema, injected
```
(I would try it myself, but the build system is a little complex for a newbie like me)

**Comment by scottbelden** [comment]
Could you give an example of a failure?

**Comment by majensen** [comment]
Sure - Try with this set of avsc: https://gist.github.com/majensen/e75b9f1cef405dc5a1cbb896c6f5f8df
fastavro.schema.load_schema("Parent.avsc") succeeds with 1.1.1 and fails with 1.2.2

**Comment by scottbelden** [comment]
@majensen Thanks again for the example. I think I have it fixed in PR #492. If you have a moment, could you test using that code? It works for your example but I just wanted to see if there were any other issues that your schema runs into.

To test you should be able to [grab the wheel here](https://github.com/fastavro/fastavro/actions/runs/441500816) and then install it with something like this:

```
pip uninstall fastavro
pip install --no-index -f <download_folder> fastavro
```

If you don't have time to test then that's okay. I'll still plan to merge and release it sometime soon as it definitely is a bug.

**Comment by majensen** [comment]
Hey @scottbelden - really appreciate your working on this. PR #492 seems to do the trick with the real (more complicated) schema - all the tests pass.
(turns out I could point pip at the wheel directly: `pip install fastavro-..._x86_64.whl`)
